org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/AnalyzeProjectJob.java

```diff
@@ -337,17 +343,15 @@ public void run() {
           @Override
           public void handle(Issue issue) {
             IResource r;
-            if (issue.getFilePath() == null) {
+            if (issue.getInputFile() == null) {
               r = request.getProject();
             } else {
-              r = ResourcesPlugin.getWorkspace().getRoot().getFileForLocation(Path.fromOSString(issue.getFilePath().toString()));
+              r = issue.getInputFile().getClientObject();
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) NullPointerException might be thrown as 'issue' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2259)